### PR TITLE
remove unnecessary classes, exceptions ref - all custom now, dtos to Data

### PR DESCRIPTION
**Pull request by kolesroma**
still working on task 85

### CODE REVIEW

**Comment by kolesroma** [comment]
we can create our own annotation for dto @ Dto

**Review by maximelyashevich — APPROVED** [review]
Hi @kolesroma,
Approved from my side - good approach with custom supplier!

Regards,
Maks

File: src/main/java/com/epam/emotionalhelp/service/impl/RecommendationServiceImpl.java [hunk 1]
```diff
@@ -16,34 +21,39 @@
...
+                    .build();
+        }
     }
 
-    @Override
-    public List<RecommendationEntity> findAllByEmotionId(Long emotionId) {
-        return recommendationRepository.findAllByEmotionId(emotionId);
+    private Predicate<RecommendationEntity> isInsideInterval(EmotionDto em) {
+        return rec -> rec.getFloor() <= em.getValue() && em.getValue() <= rec.getCeil();
     }
```
Comment (maximelyashevich) [inline src/main/java/com/epam/emotionalhelp/service/impl/RecommendationServiceImpl.java:58]:
```suggestion
    private Predicate<RecommendationEntity> isInsideInterval(EmotionDto emotion) {
        final int value = emotion.getValue();
        return return recommendation -> recommendation.getFloor() <= value && value <= recommendation.getCeil();
    }
```

**Review by maximelyashevich — APPROVED** [review]
Hi @kolesroma,
Approve from my side - excellent work! Please consider my comments to improve code quality - and you can close this PR.

Regards,
Maks

**Pull request merged**